fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -866,6 +867,20 @@ RecordCursor<FDBStoredRecord<M>> scanRecords(@Nullable Tuple low, @Nullable Tupl
                                                  @Nullable byte[] continuation,
                                                  @Nonnull ScanProperties scanProperties);
 
+    /**
+     * Scan the records in the database in a key range.
+     *
+     * @param range key range
+     * @param continuation any continuation from a previous scan
+     * @param scanProperties skip, limit and other scan properties
+     *
+     * @return a cursor that will scan everything in the range, picking up at continuation, and honoring the given scan properties
+     */
+    @Nonnull
+    RecordCursor<FDBStoredRecord<M>> scanRecordsKeyRange(@Nonnull KeyRange range,
```

Comment:
Do you think this API should be `INTERNAL`? Most of the use cases I can think of are probably internal, though there may be some use case for APIs that returns a `KeyRange` that the user can hand back, or something.